pylint/config/argument.py

```diff
@@ -80,6 +81,17 @@ def _non_empty_string_transformer(value: str) -> str:
     return pylint_utils._unquote(value)
 
 
+def _path_transformer(value: str) -> str:
+    """Check that a path is valid and expand user / variable."""
+    # Expand tilde to allow spelling-private-dict-file = ~/.pylintdict
+    value = os.path.expanduser(value)
+    value = os.path.expandvars(value)
+    # We'd need to distinguish between input path that should exist and output path that do not
+    # if not os.path.exists(value):
+    #     raise argparse.ArgumentTypeError(f"'{value}' is not a valid path is does not exists.")
```

Comment:
@DanielNoord what do you think, is it worth it ?

Reply:
Probably not, as I can imagine we might also want to use this as an `output` path at some point. So then it not existing is not a problem.

I think the `OSError` should probably be caught in the checkers themselves.